src/misc/setup.cpp

```diff
@@ -1248,56 +1250,84 @@ bool Config::ParseConfigFile(const std::string& type, const std::string& configf
 		return false;
 	}
 
-	configfiles.push_back(configfilename);
+	configfiles.push_back(config_file_name);
 	configFilesCanonical.push_back(canonical_path);
 
-	// Get directory from configfilename, used with relative paths.
+	// Get directory from config_file_name, used with relative paths.
 	current_config_dir = canonical_path.parent_path().string();
 
-	string line;
-	Section* currentsection = nullptr;
+	// If this is an autoexec section, the above takes care of the joining
+	// while this handles the overwrriten mode. We need to be prepared for
+	// either scenario to play out because we won't know the users final
+	// preference until the very last configuration file is processed.
+
+	std::string line = {};
+
+	Section* current_section = nullptr;
+	bool is_autoexec_section = false;
+	bool is_autoexec_started = false;
+
+	auto is_empty_line = [](const std::string& line) {
+		return line.empty() || line[0] == '\0' || line[0] == '\n' ||
+		       line[0] == '\r';
```

Comment:
Are tabs stripped off prior to this (and so we don't need to check for them)?

Reply:
The `trim(line)` call strips away all the leading/trailing spaces/tabs from each and every line - just like before my change.